### PR TITLE
docs: Move Effects Examples

### LINKED ISSUE #1949: Interactive docs: `MoveEffect`
# Interactive doc for the `MoveEffect`s

The three `MoveEffect`s should have small examples that show how the effect is applied and how it works. This could for example be just clicking somewhere in the game and moving a geometrical shape to the position where the reader clicked.

https://docs.flame-engine.org/1.3.0/flame/effects.html#movebyeffect
https://docs.flame-engine.org/1.3.0/flame/effects.html#movetoeffect
https://docs.flame-engine.org/1.3.0/flame/effects.html#movealongpatheffect

# What are interactive docs?

On our documentation site has the possibility to run inline interactive examples and we would like every visual feature to have a **small**, pretty and easily understandable example attached to it. Some current examples are the [router](https://docs.flame-engine.org/1.3.0/flame/router.html) and the [decorators](https://docs.flame-engine.org/1.3.0/flame/rendering/decorators.html).
Some inspiration can also be found on our [examples page](https://examples.flame-engine.org) where we have examples for most features.

# Good to know

Make sure that you comment on the issue that you will be working on this before starting to program.

To set up the Flame environment you need to follow the instructions in our [contributing.md](https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md#environment-setup) doc, if you have any problems setting up your environment (or other issues of course) feel free to ask for help in the Flame channel on our [Discord server](https://discord.com/invite/pxrBmy4).

Since this is a documentation change, you need to be able to build the docs. Instructions for building the documentation site locally can be found here: https://docs.flame-engine.org/main/development/documentation.html#building-documentation-locally

Make sure to include a video of the example when you upload the PR so that it will be easier for us to review it. (You can attach videos directly in the PR, so no need to upload them somewhere else)

**Happy hacking!**

### PR DESCRIPTION
**Pull request by sccalabr**
# Description
The three MoveEffects should have small examples that show how the effect is applied and how it works. This could for example be just clicking somewhere in the game and moving a geometrical shape to the position where the reader clicked.

https://docs.flame-engine.org/1.3.0/flame/effects.html#movebyeffect
https://docs.flame-engine.org/1.3.0/flame/effects.html#movetoeffect
https://docs.flame-engine.org/1.3.0/flame/effects.html#movealongpatheffect

## Checklist

- [X ] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [ X] I have followed the [Contributor Guide] when preparing my PR.
- [ ]X I have updated/added tests for ALL new/updated/fixed functionality.
- [ X] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [ X] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [X ] No, this PR is not a breaking change.

## Related Issues

Closes #1949

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

**Comment by sccalabr** [comment]
[screen-recording.webm](https://user-images.githubusercontent.com/4111230/194732088-adb02570-41df-4aef-91f4-1770db56fa7e.webm)

**Comment by munsterlander** [comment]
@sccalabr As I mentioned in the issue:

- Use the naming convention already established for file and class names. `MoveToEffectGame` `move_to_effect.dart` not `effects_moveto.dart`, etc
- The other sections use the factory: MoveEffect.to and not MoveToEffect.  For consistency, consider making everything the same in the docs.
- You are creating the effect and applying it after when you will be better served to create the effect in the onTap.
- Reset the effect so users do not need to refresh the whole page to see what happens again.
- Update the `effects.md` file so your examples are multiline and don't force mobile users to scroll.
- Please follow: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
  - Format the code using flutter format .;
  - Lint the code with melos run analyze;
  - Check that all tests pass: melos run test;

**Comment by sccalabr** [comment]
[screen-recording(1).webm](https://user-images.githubusercontent.com/4111230/194735264-2522724a-57d0-41bd-9e66-9a35f998fbca.webm)

**Comment by sccalabr** [comment]
[screen-recording(2).webm](https://user-images.githubusercontent.com/4111230/194735357-22fec149-db51-4d85-94ee-2dbf3d5132ac.webm)

**Comment by sccalabr** [comment]
i think i got it working :)

**Comment by munsterlander** [comment]
Yup, looks good to me. Just like the code that was in my branch. 😀

File: doc/flame/examples/lib/move_along_path_effect.dart [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+          flower.add(
+            MoveAlongPathEffect(
+              Path()..quadraticBezierTo(100, 0, 50, -50),
+              EffectController(duration: 1.5),
+            ),
+          );
+        } else {
+          flower.add(
+            MoveEffect.by(
+              Vector2(size.x / 2, size.y / 2),
```
Comment (spydon) [inline doc/flame/examples/lib/move_along_path_effect.dart:26]: This can be just `size / 2`

File: doc/flame/examples/lib/move_by_effect.dart [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+          flower.add(
+            MoveEffect.by(
+              Vector2(30, 30),
+              EffectController(duration: 1.0),
+            ),
+          );
+        } else {
+          flower.add(
+            MoveEffect.by(
+              Vector2(size.x / 2, size.y / 2),
```
Comment (spydon) [inline doc/flame/examples/lib/move_by_effect.dart:24]: Same here

**Review by sccalabr — COMMENTED** [review]
addressing comments

File: doc/flame/examples/lib/move_along_path_effect.dart (hunk 1 above)
Comment (sccalabr) [reply doc/flame/examples/lib/move_along_path_effect.dart]: looks like the build fails when doing that

Comment (sccalabr) [reply doc/flame/examples/lib/move_along_path_effect.dart]: The argument type 'Vector2' can't be assigned to the parameter type 'double'

**Review by sccalabr — COMMENTED** [review]
reverting back to what it was because of this error

The argument type 'Vector2' can't be assigned to the parameter type 'double'

File: doc/flame/examples/lib/move_along_path_effect.dart (hunk 1 above)
Comment (spydon) [inline doc/flame/examples/lib/move_along_path_effect.dart:26]:
```suggestion
              size / 2,
```
It should simply be like this :point_up:

**Review by spydon — APPROVED** [review]

**Comment by spydon** [comment]
> From what i am seeing in the documentation. The move effects are not reset or reversed properly. 2 out of 3 are going out of the box when interacting for 2nd time.

Good catch, do you want to send a PR that fixes it?

**Comment by immadisairaj** [comment]
> Good catch, do you want to send a PR that fixes it?

Yes, I can send a PR for it. Does it need an issue or I can directly send a PR?

**Comment by spydon** [comment]
> > Good catch, do you want to send a PR that fixes it?
> 
> Yes, I can send a PR for it. Does it need an issue or I can directly send a PR?

PR directly :)

**Pull request merged**